libcontainer/container_linux.go

```diff
@@ -169,21 +172,65 @@ func (c *linuxContainer) newInitProcess(p *Process, cmd *exec.Cmd, parentPipe, c
 			cmd.SysProcAttr.Credential = &syscall.Credential{}
 		}
 	}
-	cmd.Env = append(cmd.Env, t)
-	cmd.SysProcAttr.Cloneflags = cloneFlags
+	// if we required to create a new user namespace, delegates to golang
+	// implementation to be able to set uid/gid mappings in a standard way,
+	// otherwise do it inside nsexec by passing the clone flags because we dont
+	// have to perform any additional setup when start a new process.
+	if c.config.Namespaces.PathOf(configs.NEWUSER) == "" {
+		cmd.SysProcAttr.Cloneflags = cloneFlags
```

Comment:
If you need to create a new user namespace and attach to a precreated pid namespace, you need to enter into the pid namespace and only then create a user namespace, so the user namespace should be created from nsexec.c, should not it?